src/katello/agent/pulp/libdnf.py

```diff
@@ -104,8 +111,8 @@ def __init__(self, transaction):
         self.resolved = []
         self.failed = []
         for item in transaction:
-            po = item.installed or item.erased
-            if po:
+            po = item.pkg
+            if item.action in self.ACTION_LIST:
```

Comment:
I think this code was meant to track the state of things after the transaction runs however the RPMTransaction is getting cleared out after the fact by DNF. Because the report is generated after the txn runs nothing meaningful is sent back to the server. ex: "No new packages installed" when packages were actually installed

Might be able to divine the way to do this as intended from dnf.Base code, nothing is jumping out at me right now

Reply:
@jturel were you seeing "No new packages installed" when packages were installed?  For me it properly reported the list of packages installed/updated/removed

Reply:
Fixed after updating dnf & libdnf.

previous: dnf-4.0.4-1.el8.noarch libdnf-0.22.0-2.el8.x86_64
now: dnf-4.0.9.2-5.el8.noarch libdnf-0.22.5-4.el8.x86_64

Reply:
I still wonder about this code perhaps even before your changes. I don't see how lists could be an accurate representation of what happened b/c I don't think the RPMTransaction actually reflects post-transaction state.

I think the below would map more closely to the way it was written: "if this transaction item was installed or erased, add to resolved, otherwise add to failed"

```
installed_or_removed = transaction.install_set + transaction.remove_set
for item in transaction:
  if item.pkg in installed_or_removed:
      _list = self.resolved
  else:
       _list= self.failed
 .....
```
I'm not really sure if it's accurate though.

Looking at the code here: https://github.com/rpm-software-management/dnf/blob/35d1c3b40ff533a8a2afadeee956ea26825d9949/dnf/db/group.py#L323

It looks like a RuntimeError would be raised where there was a transaction item without a package

Also, if something went wrong in the txn, an error would be raised before this report is generated:

https://github.com/rpm-software-management/dnf/blob/3c88bc6d39a965d93452188de88e63cea7225d11/dnf/base.py#L863

Reply:
updated based on discussions

Reply:
It might be better to check for the failed state first because I think it could be possible for the item.action to be in ACTION_LIST but for the item.state to be fail-ey. Even if that's not the case now it might guard against future changes in dnf behavior

Reply:
Sounds good, updated!